wgpu/src/lib.rs

```diff
@@ -3192,8 +3202,18 @@ impl<'a> RenderPass<'a> {
 
     /// Draws indexed primitives using the active index buffer and the active vertex buffers.
     ///
-    /// The active index buffer can be set with [`RenderPass::set_index_buffer`], while the active
-    /// vertex buffers can be set with [`RenderPass::set_vertex_buffer`].
+    /// The active index buffer can be set with [`RenderPass::set_index_buffer`]
+    /// The active vertex buffers can be set with [`RenderPass::set_vertex_buffer`].
+    ///
+    /// Panics if indices Range is not within 0..BufferLen / BufferStride.
```

Comment:
as above
ideally also contains again a note on vertex range requirements (although I don't even know right now what is supposed to happen when an index is out of bounds for a vertex buffer - I know this is detected, but not sure if it's an error or defined as zeroed data)

Reply:
a valid program should also never use vertex indices with [GPUVertexStepMode."vertex"](https://www.w3.org/TR/webgpu/#dom-gpuvertexstepmode-vertex) that are out of bounds. WebGPU implementations have different ways of handling this, and therefore a range of behaviors is allowed. Either the whole draw call is discarded, or the access to those attributes out of bounds is described by WGSL’s [invalid memory reference](https://gpuweb.github.io/gpuweb/wgsl/#invalid-memory-reference).

is how they describe this issue. It could be a error or a discarded draw.